osu.Game/Utils/IDeepCloneable.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Utils
+{
+    /// <summary>A generic interface for a deeply cloneable type.</summary>
+    /// <typeparam name="T">The type of object to clone.</typeparam>
+    public interface IDeepCloneable<out T> where T : class
```

Comment:
Is this interface going to be used in a generic code? It seems unnecessary.

Reply:
Not for now. The premise for adding this is to provide xmldoc to each implementation, and at least keep conformity of implementation (as until now we had deep clone methods in many places intending to do the same thing but not linked in any way).